Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.mm

```diff
@@ -38,14 +38,29 @@
 WebViewRenderingUpdateScheduler::WebViewRenderingUpdateScheduler(WebView* webView)
     : m_webView(webView)
 {
+    ASSERT(isMainThread());
     ASSERT_ARG(webView, webView);
 
-    m_renderingUpdateRunLoopObserver = makeUnique<WebCore::RunLoopObserver>(WebCore::RunLoopObserver::WellKnownOrder::RenderingUpdate, [this] {
-        this->renderingUpdateRunLoopObserverCallback();
+    m_renderingUpdateRunLoopObserver = makeUnique<WebCore::RunLoopObserver>(WebCore::RunLoopObserver::WellKnownOrder::RenderingUpdate, [weakThis = WeakPtr { this }] {
+#if PLATFORM(IOS_FAMILY)
+    // Normally the layer flush callback happens before the web lock auto-unlock observer runs.
+    // However if the flush is rescheduled from the callback it may get pushed past it, to the next cycle.
```

Comment:
Nit - indentation seems inconsistent here (though it could just be the GitHub diff view)

Reply:
Oh oops, fixed.